### ISSUE TITLE
[BUG]Error: The argument type 'PointerEvent' can't be assigned to the parameter type 'PointerDownEvent'

### ISSUE THREAD

**Issue by rafaelgs2311** [bluefireteam/photo_view#459]
Error: The argument type 'PointerEvent' can't be assigned to the parameter type 'PointerDownEvent'.

Error generated on build progress...
Its necessary to change void addAllowedPointer(PointerEvent event) to void addAllowedPointer(PointerDownEvent event) 
according https://flutter.dev/docs/release/breaking-changes/gesture-recognizer-add-allowed-pointer

**Comment by kamleshwebtech** [comment]
any solution?? Kindly share. Thanks a lot.

**Comment by SudhindraAnegondhi** [comment]
This is due to a  Flutter 2.5 Breaking Change. [GestureRecognizer cleanup](https://flutter.dev/docs/release/breaking-changes/gesture-recognizer-add-allowed-pointer).

The change required is minor and until it is fixed, I would advise reverting to Flutter 2.3

**Comment by mirshahbazi** [comment]
how can fix this issue? why did you close this issue!

**Comment by Kaffeetasse** [comment]
@mirshahbazi you could fix it by using the version from this pull request #462

**Comment by mirshahbazi** [comment]
> @mirshahbazi you could fix it by using the version from this pull request #462

I use native_pdf_view: ^4.1.0 this library use  photo_view: ^0.11.0, now how can i update that?

**Comment by mirshahbazi** [comment]
I used this code and my problem was solved 💯 

```
dependency_overrides:
  photo_view:
    # custom photo_view with better gesture recognition
    git: git://github.com/robertodoering/photo_view.git
```

**Comment by prilepskiy** [comment]
> I used this code and my problem was solved 💯
> 
> ```
> dependency_overrides:
>   photo_view:
>     # custom photo_view with better gesture recognition
>     git: git://github.com/robertodoering/photo_view.git
> ```

But I have version solving problem with this:

```
Because tweet_ui 3.0.0 depends on photo_view ^0.11.1 and no versions of tweet_ui match >3.0.0 <4.0.0, tweet_ui ^3.0.0 requires photo_view from hosted.
So, because meduza depends on both tweet_ui ^3.0.0 and photo_view from git, version solving failed.
Running "flutter pub get" in meduza-dart...
pub get failed (1; So, because meduza depends on both tweet_ui ^3.0.0 and photo_view from git, version solving failed.)
```

**Comment by jagomf** [comment]
As a temporary workaround (while a fixed version is released), just one simple change is required to your hosted photo_view plugin:

1. Go to your local flutter environment path (in windows it is `C:\flutter`)
2. Go to `\.pub-cache\hosted\pub.dartlang.org\photo_view-0.11.1\lib\src\core\`
3. Edit `photo_view_gesture_detector.dart`:

Replace in **line 101**:
```
void addAllowedPointer(PointerEvent event) {
```
with
```
void addAllowedPointer(PointerDownEvent event) {
```

That should be enough to be able to build again.

Thanks to @Kaffeetasse for pointing out @karvulf PR.

**Comment by renancaraujo** [comment]
This was supposed to be fixed on 0.12.0. Which version of photo_view are you using?

**Comment by renancaraujo** [comment]
Can anyone reproduce this error when using 0.12.0?

**Comment by prilepskiy** [comment]
Sorry, I can't update to 0.12.0 because of version solving error with tweet_ui package, I'm waiting for [this PR](https://github.com/schibsted/tweet_ui/pull/71)

**Comment by renancaraujo** [comment]
I am closing since this is supposed to be fixed already on 0.12.0. 

 If you have any version solving problems with another package, do this:
 
 - Ask the maintainer of the package to update photo view.
 - Add this to your pubspec meanwhile: 
 ```
 dependency_overrides:
  photo_view: 0.12.0
 ```

**Comment by prilepskiy** [comment]
> * Add this to your pubspec meanwhile:
> 
> ```
> dependency_overrides:
>  photo_view: 0.12.0
> ```
Yes, it helped. 
And I can't reproduce this error when using 0.12.0, thank you! 🙏
